cloudinit/config/cc_users_groups.py

```diff
@@ -173,6 +173,15 @@ def handle(name, cfg, cloud, _log, _args):
             else:
                 config["ssh_redirect_user"] = default_user
                 config["cloud_public_ssh_keys"] = cloud_keys
+
+        if (
+            config.get("no_create_home") or config.get("system")
+        ) and config.get("ssh_authorized_keys"):
```

Comment:
I just realized after testing point 4 above, we still need to avoid the errant creation of the homedir due to ssh_authkey_fingerprints... so if we can add a log message there about the skip of printing. I'm satisfied. Thanks again

Reply:
ok. I've rebased and then added commits to the PR. Please see if this is ok?

Reply:
@blackboxsw is this ok for you?